src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Formatting/CSharpFormatter.cs

```diff
@@ -262,11 +262,17 @@ private static (Dictionary<int, IndentationMapData>, SyntaxTree) InitializeInden
 
         using var changes = new PooledArrayBuilder<TextChange>();
 
+        var syntaxTree = CSharpSyntaxTree.ParseText(context.CSharpSourceText, cancellationToken: cancellationToken);
+        var root = syntaxTree.GetRoot(cancellationToken);
+
         var previousMarkerOffset = 0;
         foreach (var projectedDocumentIndex in projectedDocumentLocations)
         {
-            var useMarker = char.IsWhiteSpace(context.CSharpSourceText[projectedDocumentIndex]);
-            if (useMarker)
+            var token = root.FindToken(projectedDocumentIndex, findInsideTrivia: true);
+
+            // We use a marker if the projected location is in trivia, because we can't add annotations to a specific piece of trivia
+            var isInTrivia = projectedDocumentIndex < token.SpanStart || projectedDocumentIndex >= token.Span.End;
```

Comment:
Do we need to check if the trivia is something other than whitespace?

Reply:
I think I might have just misunderstood what was going on. At first, I thought this was specifically looking for whitespace, not all trivia